### PR TITLE
Enhance error handling and response wrapping in AI message processing

### PR DESCRIPTION
**Pull request by orgaralf**
## Summary
These changes aim to improve the overall user experience by providing clearer error messages and more reliable response handling.

## Changes
- Updated `FailureClassifier` to include new subscription-related error messages, improving clarity for users regarding model availability.
- Enhanced `GatewayToolLoop` to better manage tool choice conflicts and wrap responses when no valid URLs are present, ensuring smoother user interactions.
- Improved `GeminiMessagesTranslator` and `OpenAiMessagesTranslator` to handle thought signatures more effectively across function calls, maintaining consistency in message processing.
- Added tests to validate new functionality and ensure robust handling of edge cases in message translations and tool interactions.

## Verification
- [ ] Not tested (explain)
- [X] Manual
- [ ] Tests added/updated

### CODE REVIEW

File: backend/src/AI/Health/FailureClassifier.php [hunk 1]
```diff
@@ -56,6 +56,11 @@
         'is sunset',
         'sunsetting',
         'not_found_error',
+        // Provider plan: this one model is off the current key, others still work.
+        'subscription tier',
+        'not available in your subscription',
+        'not included in your plan',
+        'requires a higher plan',
```
[the remark this answers is not in the document]
Comment (orgaralf) [reply backend/src/AI/Health/FailureClassifier.php]: Agreed. Bare “subscription tier / higher plan” wording is account-wide and must not retire one model.

`f97cdadc9` keeps Permanent only for model-scoped phrases (`model is not available in your subscription`, …). Account-wide strings now classify as Credential, same as billing. Tests cover expiry and “requires a higher plan”.

File: backend/src/Service/File/VectorizationService.php [hunk 2]
```diff
@@ -209,11 +222,14 @@ public function vectorizeAndStore(
                 );
 
                 ++$chunksCreated;
+                if (\count($pending) >= self::EMBED_BATCH_SIZE) {
+                    $this->vectorStorage->storeChunkBatch($pending);
+                    $pending = [];
```
[the remark this answers is not in the document]
Comment (orgaralf) [reply backend/src/Service/File/VectorizationService.php:234]: Fixed. Each `storeChunkBatch()` still commits a window (needed to stay under memory), but if a later window throws we now `deleteByFile()` for this invocation so the file is not left half-indexed.

Covered by `testPartialStoreIsRolledBackWhenALaterWindowFails`.

File: backend/src/AI/Messages/Tools/GatewayToolLoop.php [hunk 3]
```diff
@@ -318,11 +349,24 @@ public function runStream(
             $totalUsage = $this->sumUsage($totalUsage, $turn['usage']);
 
             if ($turn['error']) {
+                if ($this->alreadyWrapped($body) && $webSearchRounds > 0) {
+                    $emitter->emitAssistantText($this->recoverWrapUpText($body, $webSearchRounds));
+                    $this->finishStream($emitter, ['held_stop' => null]);
+
+                    return $totalUsage->withStopReason('end_turn');
```
[the remark this answers is not in the document]
Comment (orgaralf) [reply backend/src/AI/Messages/Tools/GatewayToolLoop.php]: Fixed. Wrap-up recovery now suppresses the upstream `error` SSE (`relayErrors: false`) and only then emits the recovered assistant text + `end_turn`. The client no longer sees error-then-success on the same stream.

Recovery is also gated on `$webSearchRounds > 0`, so a coincidental wrap-up prompt without a search does not 200 an empty-search answer.

File: backend/src/AI/Messages/Tools/GatewayToolLoop.php [hunk 4]
```diff
@@ -973,6 +1026,95 @@ private function forceWrapUp(array $requestBody): array
...
+
+    /**
+     * @param array<string, mixed> $requestBody
+     */
+    private function recoverWrapUpText(array $requestBody, int $webSearchRounds): string
+    {
+        if ($webSearchRounds < 1) {
+            return self::EMPTY_SEARCH_RECOVERY;
+        }
+        $urls = $this->httpUrlsFromMessages($requestBody['messages'] ?? []);
```
[the remark this answers is not in the document]
Comment (orgaralf) [reply backend/src/AI/Messages/Tools/GatewayToolLoop.php]: Fixed. Recovery now walks `tool_result` blocks only (the executed search), not the user prompt or earlier assistant turns. Tokens need a real host (`https://?` no longer counts). The Groq wrap-up test now plants `https://evil.example` in the prompt and asserts it is not cited.

File: backend/src/Service/File/VectorizationService.php [hunk 5]
```diff
@@ -72,6 +80,11 @@ public function vectorizeAndStore(
         }
 
         try {
+            // Large desktop folders (3 MB handbook) take minutes of local
+            // embeddings. A 300s max_execution_time turns that into HTTP 500
+            // and a Failed file. The worker and CLI already run without a cap.
+            set_time_limit(0);
```
[the remark this answers is not in the document]
Comment (orgaralf) [reply backend/src/Service/File/VectorizationService.php]: Fixed. Dropped the one-shot `set_time_limit(0)`. Each embed/store window now re-arms a 120s budget via `function_exists('set_time_limit')`, same pattern as the long-running provider poll loops.
